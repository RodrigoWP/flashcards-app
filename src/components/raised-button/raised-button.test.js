import React from 'react'
import { shallow } from 'enzyme'
import { RaisedButton } from '.'

it('Should be render correctly', () => {
    const Component = shallow(
        <RaisedButton 
            label='button'
            onPress={jest.fn} />
    )

    expect(Component).toMatchSnapshot()
})

it('Should be call onPress when button is clicked', () => {
    const onPress = jest.fn()

    const component = shallow(
        <RaisedButton
            label='button'
            onPress={onPress} />
    )

    const render = component.find('TouchableHighlight')
    render.simulate('press')

    expect(onPress).toHaveBeenCalledTimes(1)
})

it('Should not be call onPress when button is clicked if disable', () => {
    const onPress = jest.fn()

    const component = shallow(
        <RaisedButton
            label='button'
            onPress={onPress}
            disable />
    )

    const render = component.find('TouchableHighlight')
    render.simulate('press')

    expect(onPress).toHaveBeenCalledTimes(0)
})
