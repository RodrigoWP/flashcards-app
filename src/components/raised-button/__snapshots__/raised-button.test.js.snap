// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be render correctly 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "height": 50,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#6197c6",
      },
      undefined,
    ]
  }
  underlayColor="#afcef3"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
  >
    BUTTON
  </Text>
</TouchableHighlight>
`;
